synapse/app/homeserver.py

```diff
@@ -16,10 +16,23 @@
 
 import sys
 sys.dont_write_bytecode = True
-from synapse.python_dependencies import check_requirements, DEPENDENCY_LINKS
+from synapse.python_dependencies import (
+    check_requirements, DEPENDENCY_LINKS, MissingRequirementError
+)
 
 if __name__ == '__main__':
-    check_requirements()
+    try:
+        check_requirements()
```

Comment:
That might be more confusing than helpful since the pynacl being the wrong version of 0.3.0 caused the pymacaroons import to fail. If people are presented with too many failures they might end up not reinstalling the right packages.